### PR TITLE
Fix conversion for UpdateStatus from the gRPC layer

### PR DESCRIPTION
**Pull request by agourlay**
This PR fixes a nasty bug happening when remote shards use the wait flag to `false`.

It is due to an incomplete mapping of the gRPC `UpdateStatus` to the operations `UpdateStatus`.

Here is the mapping code.

```rust
impl From<UpdateResult> for api::grpc::qdrant::UpdateResult {
    fn from(value: UpdateResult) -> Self {
        Self {
            operation_id: value.operation_id,
            status: value.status as i32,
        }
    }
}

impl TryFrom<api::grpc::qdrant::UpdateResult> for UpdateResult {
    type Error = Status;

    fn try_from(value: api::grpc::qdrant::UpdateResult) -> Result<Self, Self::Error> {
        Ok(Self {
            operation_id: value.operation_id,
            status: match value.status {
                1 => UpdateStatus::Acknowledged,
                2 => UpdateStatus::Completed,
                _ => return Err(Status::invalid_argument("Malformed UpdateStatus type")),
            },
        })
    }
}
```

The first conversion relies on the C-style enum however, the enum is missing the `0` entry.

I have improved the logging as it was difficult to find this error.

### CODE REVIEW

File: lib/collection/src/operations/types.rs [hunk 1]
```diff
@@ -88,6 +88,8 @@ pub struct CollectionInfo {
 #[derive(Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
 #[serde(rename_all = "snake_case")]
 pub enum UpdateStatus {
+    /// Unknown update status
+    Unknown,
```
Comment (generall) [inline lib/collection/src/operations/types.rs:92]: This is going to change the API. Maybe we better add another match to the conversion instad?

File: lib/collection/src/operations/types.rs [hunk 2]
```diff
@@ -385,13 +387,13 @@ impl From<tonic::Status> for CollectionError {
     fn from(err: tonic::Status) -> Self {
         match err.code() {
             tonic::Code::InvalidArgument => CollectionError::BadInput {
-                description: "InvalidArgument".to_string(),
+                description: format!("InvalidArgument: {}", err),
```
Comment (e-ivkov) [inline lib/collection/src/operations/types.rs:390]: If you'd like, you can change them like this:
```suggestion
                description: format!("InvalidArgument: {err}"),
```

Comment (generall) [reply lib/collection/src/operations/types.rs:390]: let's maybe do it some in some other pr, cause there are currently all other code use the first version

**Review by e-ivkov — APPROVED** [review]

**Comment by agourlay** [comment]
Thanks for the change, I like it better :+1:

**Comment by e-ivkov** [comment]
Changed the example there, the syntax was a bit off

**Comment by agourlay** [comment]
@e-ivkov Thank you for you suggestion, I applied it in https://github.com/qdrant/qdrant/pull/803/commits/d906b944d09c6b4ef04143210a2d921f53f57707

I will merge on green.

**Pull request merged**